cloud/services/loadbalancers.go

```diff
@@ -114,6 +114,11 @@ func AddNodeToNB(
 	if !kutil.IsControlPlaneMachine(machineScope.Machine) {
 		return nil
 	}
+
+	if machineScope.LinodeMachine.Spec.InstanceID == nil {
+		return errors.New("no InstanceID set for LinodeMachine.Spec")
+	}
```

Comment:
Is `InstanceID` required? According to https://github.com/linode/cluster-api-provider-linode/blob/main/api/v1alpha1/linodemachine_types.go#L35-L37 it's optional. If it's required, we should set the validation tags in the API, not here. This way we get field validation for free from Kubernetes.

Reply:
Ah good catch. I didn't realize we set it to optional. I added that condition as I was working through tests and thought why don't we check this variable?

I'll remove that :)